engine/common/hpak.c

```diff
@@ -18,7 +18,7 @@ GNU General Public License for more details.
 
 #define HPAK_MAX_ENTRIES	0x8000
 #define HPAK_ENTRY_MIN_SIZE	(512)
-#define HPAK_ENTRY_MAX_SIZE	(128 * 1024)
+#define HPAK_ENTRY_MAX_SIZE	(512 * 1024)
```

Comment:
128 kB still not enough, but 256 kB will be enough according "expand limit as less as possible" logic

Reply:
Not enough for what exactly? Can you provide an example?

Player decal texture limit is 512x512. I think most sane images, not filled with noise, hitting that limit, can be compressed into 128k. Maybe even indexed or crushed into lesser pixel format, but I think we're talking about like clan logos here.

Reply:
> Player decal texture limit is 512x512.

We can't even reach that, because of total size limit. For example, I have image with 125 kB total size, and resolution are 378x420 - it just can't be raised more. In reality, PNG doesn't compresses too much, unlike JPEG does.

Reply:
Did you try to recompress it with pngcrush/optipng/pngquant?

pngquant for example does an excellent job recompressing stuff most of the time.

Reply:
Explicitly, no. Maybe GIMP includes it and uses under the hood, I was using maximum compression level for exporting PNG in there. Anyway, problem with all such external tools it that almost nobody uses something except generic photo editors like Paint/GIMP/Photoshop/built-in OS photo editor.

Reply:
No, implicitly nothing uses it as far as I know.
Yet, I'm not going to support random software quirks.

There is a limit, there is support for TGA, BMP and PNG, TGA and BMP support palette, TGA supports RLE encoding, for PNG there are tools like pngquant, just get creative.

Reply:
@nekonomicon wrong thread :)

zopflipng didn't do much on the image above. Didn't have enough patience with it to enable bruteforce enough.